languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -192,24 +199,27 @@ public void registerCall(String attrName, List<Object> args, long callId, JSONOb
         } else {
             instance = host.toJava(polarInstance);
         }
-
         // Select a method to call based on the types of the arguments.
-        Class<?>[] argTypes = args.stream().map(a -> a.getClass())
-            .collect(Collectors.toUnmodifiableList())
-            .toArray(new Class[0]);
         Object result = null;
         try {
             Class<?> cls = instance instanceof Class ? (Class<?>) instance : instance.getClass();
-            try {
-                Method method = cls.getMethod(attrName, argTypes);
-                result = method.invoke(instance, args.toArray());
-            } catch (NoSuchMethodException e) {
+            if(args.isPresent()) {
+                Class<?>[] argTypes = args.get().stream().map(a -> a.getClass())
+                    .collect(Collectors.toUnmodifiableList())
+                    .toArray(new Class[0]);
+                try {
+                    Method method = cls.getMethod(attrName, argTypes);
+                    result = method.invoke(instance, args.get().toArray());
+                } catch (NoSuchMethodException e) {
+                    throw new Exceptions.InvalidCallError(cls.getName(), attrName, argTypes);
+                }
+            } else {
                 // Look for a field with the given name.
                 try {
                     Field field = cls.getField(attrName);
                     result = field.get(instance);
                 } catch (NoSuchFieldException f) {
-                    throw new Exceptions.InvalidCallError(cls.getName(), attrName, argTypes);
+                    throw new Exceptions.InvalidCallError(cls.getName(), attrName);
```

Comment:
Feels like this should be a distinct type of error, something like `Exceptions.InvalidFieldAccessError`

Reply:
Agreed, just did the same in TS.